src/SourceMap.js

```diff
@@ -3,11 +3,28 @@ const lineCounter = require('./utils/lineCounter');
 
 class SourceMap {
   constructor(mappings, sources) {
-    this.mappings = mappings || [];
+    this.mappings = this.purifyMappings(mappings) || [];
     this.sources = sources || {};
     this.lineCount = null;
   }
 
+  purifyMappings(mappings) {
+    if (mappings && mappings.length) {
```

Comment:
**a)** While we're filtering, it may be worthwhile to check for `mapping` itself before accessing properties on it:

```js
return (
   mapping &&
   mapping.source &&
   ...
)
```

**b)** The above if would be a bit more clear as:
` if (Array.isArray(mappings)) {
...
`

**c)** Assuming column should always be a number, could use `typeof mapping.original.column === 'number'` for these checks.

Reply:
Should be more strict now